src/common.ts

```diff
@@ -81,10 +81,19 @@ export type ReadParameters<Input> = {
   inputs?: Input;
 };
 
+export type WriteOptions = {
+  tags?: { name: string; value: string }[];
+};
+
 export type WriteParameters<Input> = WithSigner<
   Required<ReadParameters<Input>>
 >;
 
+export type WriteArguments<T> = [
+  EvaluationParameters<WriteParameters<T>>,
```

Comment:
I'm just realizing this now, but Write interactions should not support evaluation options

Reply:
You are correct, fixing.